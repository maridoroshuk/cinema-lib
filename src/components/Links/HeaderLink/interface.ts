import { ReactNode } from 'react';

export interface IHeaderLinkProps {
  children: ReactNode;
}
