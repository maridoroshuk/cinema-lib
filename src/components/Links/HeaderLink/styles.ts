import styled from 'styled-components';

export const ListItemStyled = styled.li`
  a:after {
    content: '';
    width: 0;
    height: ${({ theme }) => theme.sizes.height[2]}px;
    color: ${({ theme }) => theme.colors.defaultBlack};
    display: block;
    background-color: ${({ theme }) => theme.colors.gold};
    transition-duration: 300ms;
    margin-top: ${({ theme }) => theme.spaces[1]}px;
  }
  a:hover:after {
    width: 100%;
  }
  a.active:after {
    width: 100%;
  }
`;
