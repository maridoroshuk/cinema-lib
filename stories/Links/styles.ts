import styled from 'styled-components';

export const Navlink = styled.a`
  text-decoration: none;
  color: ${({ theme }) => theme.colors.neutral000};
  font-size: ${({ theme }) => theme.sizes.fontSize[24]}px;
  line-height: ${({ theme }) => theme.sizes.lineHeight[36]}px;
  font-weight: ${({ theme }) => theme.sizes.fontWeight[300]};
  cursor: pointer;
  font-family: ${({ theme }) => theme.fontFamily.Poppins};
  &.active {
    color: ${({ theme }) => theme.colors.gold};
  }
`;
