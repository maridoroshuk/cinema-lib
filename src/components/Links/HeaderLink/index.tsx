import React from 'react';
import { IHeaderLinkProps } from './interface';
import { ListItemStyled } from './styles';

export const HeaderLink = ({ children }: IHeaderLinkProps) => {
  return <ListItemStyled>{children}</ListItemStyled>;
};
