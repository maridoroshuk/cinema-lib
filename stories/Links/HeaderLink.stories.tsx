import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import { HeaderLink } from '@/components/Links/HeaderLink';

import { Navlink } from './styles';

export default {
  title: 'Links/HeaderLink',
  component: HeaderLink,
} as ComponentMeta<typeof HeaderLink>;

const Template: ComponentStory<typeof HeaderLink> = (args) => (
  <HeaderLink>
    <Navlink>{args.children}</Navlink>
  </HeaderLink>
);

export const Primary = Template.bind({});

Primary.args = {
  children: 'Home',
};
