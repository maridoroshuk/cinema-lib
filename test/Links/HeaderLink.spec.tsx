import React from 'react';
import { render } from '@testing-library/react';

import { HeaderLink } from '@/components/Links/HeaderLink';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('HeaderLink', () => {
  it('renders', () => {
    const { container } = render(
      setupTestProvider(<HeaderLink>Home</HeaderLink>),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
