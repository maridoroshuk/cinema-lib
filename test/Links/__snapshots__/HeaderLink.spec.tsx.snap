// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLink renders 1`] = `
.c0 {
  list-style-type: none;
}

.c0 a:after {
  content: '';
  width: 0;
  height: 2px;
  color: #000000;
  display: block;
  background-color: #D98639;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  margin-top: 1px;
}

.c0 a:hover:after {
  width: 100%;
}

.c0 a.active:after {
  width: 100%;
}

<li
  class="c0"
>
  Home
</li>
`;
